perpVects/QFPHelpers.cpp

```diff
@@ -38,33 +38,33 @@ std::ostream& operator<<(std::ostream& os, const unsigned __int128 &i){
 }
 
 namespace FPHelpers{
-    auto
-  swap_float_int(unsigned int val) -> float{
+  float
+  swap_float_int(uint32_t val){
     return *reinterpret_cast<float*>(&val);
   }
 
-  auto
-  swap_float_int(unsigned long val) -> double{
+  double
+  swap_float_int(unsigned long val){
      return *reinterpret_cast<double*>(&val);
   }
 
-  auto
-  swap_float_int(unsigned __int128 val) -> long double{
+  long double
+  swap_float_int(unsigned __int128 val){
     return *reinterpret_cast<long double*>(&val);
   }
 
-  auto
-  swap_float_int(float val) -> unsigned int{
-    return *reinterpret_cast<unsigned int*>(&val);
+  uint32_t
+  swap_float_int(float val){
+    return *reinterpret_cast<uint32_t*>(&val);
   }
 
-  auto
-  swap_float_int(double val) -> unsigned long{
+  unsigned long
+  swap_float_int(double val){
     return *reinterpret_cast<unsigned long*>(&val);
```

Comment:
Sorry, I thought you would understand that I meant it across all of those
functions.  I should have been more explicit.  My bad.

On Wed, Aug 24, 2016 at 7:13 PM Geof Sawaya <notifications@github.com>
wrote:

> In perpVects/QFPHelpers.cpp
> <https://github.com/Geof23/QFP/pull/13#discussion_r76166894>:
>
> >    }
> >
> > -  auto
> > -  swap_float_int(double val) -> unsigned long{
> > +  unsigned long
> > +  swap_float_int(double val){
> >      return *reinterpret_cast<unsigned long*>(&val);
>
> Now ya tell me
> … <#m_2648935024909752395_>
> On Wednesday, August 24, 2016, Michael Bentley ***@***.***> wrote: In
> perpVects/QFPHelpers.cpp <#13 (comment)
> <https://github.com/Geof23/QFP/pull/13#discussion_r76166866>>: > } > > -
> auto > - swap_float_int(double val) -> unsigned long{ > + unsigned long > +
> swap_float_int(double val){ > return *reinterpret_cast<unsigned
> long*>(&val); unsigned long should be replaced with uint64_t — You are
> receiving this because you authored the thread. Reply to this email
> directly, view it on GitHub <
> https://github.com/Geof23/QFP/pull/13/files/57db3da0327a58ab91d6a9ee1c70501608c4db5b..5ba2e9d97aa1ab46b0dfea9eac382ef6a136aae3#r76166866>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/AFiXD8_AOHc-IP_0UfLxnHy8O_azkH6Pks5qjOwhgaJpZM4Jr1FF>
> .
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Geof23/QFP/pull/13/files/57db3da0327a58ab91d6a9ee1c70501608c4db5b..5ba2e9d97aa1ab46b0dfea9eac382ef6a136aae3#r76166894>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AINTNNY0FmC_l3pbd4br3zyVn5xIW0J6ks5qjOw5gaJpZM4Jr1FF>
> .
>